### DISCUSSION TITLE
Equivalent of pybind11's `dynamic_attr`: desired?

### DISCUSSION THREAD

**Discussion by inducer** [wjakob/nanobind#17] [Q&A] [answered]
In putting together https://github.com/inducer/pyopencl/pull/546, I find that I'm lacking pybind11's `dynamic_attr`. I've taken a look around, and it doesn't seem like it'd be tricky to add. I'd be happy to give that a shot, unless that's one of the actively undesired features of pybind11---please let me know. Also, if there are any un-obvious gotchas to look out for, I'd be happy to know about them.

**Comment by wjakob** [comment, marked as answer]
Yes, this is currently missing. It should be relatively easy to add - you'll need to add `dynamic_attr` to `nb_attr.h`) allocate an extra flag in `type_flags` (`nb_class.h`) and then handle it in `nb_type_new` (`nb_type.cpp`)

**Comment by inducer** [comment]
https://github.com/wjakob/nanobind/pull/22
